src/main/scala/tech/cryptonomic/conseil/metadata/MetadataService.scala

```diff
@@ -50,12 +54,25 @@ class MetadataService(config: PlatformsConfiguration,
   }
 
   // fetches table attributes
-  def getTableAttributes(path: EntityPath)(implicit apiExecutionContext: ExecutionContext): Future[Option[List[Attribute]]] = for {
-    exists <- exists(path)
-    attributes <- tezosPlatformDiscoveryOperations.getTableAttributes(path.entity)
-  } yield attributes
-    .filter { _ => exists }
-    .mapNested(attribute => transformation.overrideAttribute(attribute, path.addLevel(attribute.name)))
+  def getTableAttributes(path: EntityPath)(implicit apiExecutionContext: ExecutionContext): Future[Option[List[Attribute]]] =
+    getAttributesHelper(path)(tezosPlatformDiscoveryOperations.getTableAttributes)
+
+  // fetches table attributes without updating cache
+  def getTableAttributesWithoutUpdatingCache(path: EntityPath)(implicit apiExecutionContext: ExecutionContext): Future[Option[List[Attribute]]] =
+    getAttributesHelper(path)(tezosPlatformDiscoveryOperations.getTableAttributesWithoutUpdatingCache)
+
+  // fetches attributes with given function
+  private def getAttributesHelper(path: EntityPath)(getAttributes: String => Future[Option[List[Attribute]]])
+    (implicit apiExecutionContext: ExecutionContext): Future[Option[List[Attribute]]] = {
+    exists(path).flatMap {
+      case true =>
+      getAttributes(path.entity).map(
```

Comment:
Cosmetic: wrong indentation in these cases

Reply:
reformatted